### ISSUE TITLE
Packages versions specification not respected

### ISSUE THREAD

**Issue by trybik** [ficonsulting/RInno#95]
Specifying as in README `pkgs = c(shiny = "==1.0.5", "jsonlite", "httr")` option of `create_app` will not download and install "shiny" packages in version 1.0.5 but in the most recent version (ATM 1.1.0).

It looks like while re-factoring to download packages on `create_app` the support for packages versions got lost (it's neither in `ensure` nor in `download_packages`).

**Comment by trybik** [comment]
It looks like doing that in `download_packages` would require extracting [`package_find_repo`](https://github.com/r-lib/devtools/blob/master/R/install-version.r#L31) and [`remote_download`](https://github.com/r-lib/devtools/blob/master/R/install-remote.R#L51) functions with some surrounding code from devtools.

Probably it's easier for now to just update docs here and manage packages versions by manually maintaining the `bin/` folder contents, however this would also mean that if ones wants to keep his app in a git repo, then one needs to also keep there all the binaries, which is dreadful.

**Comment by Dripdrop12** [comment]
Yeah, we need to improve the way this works. There is a part of me that just says we should update the docs and encourage people to update to CRAN versions of packages while developing their apps.

We may need to drop support for specific package versions because binaries are not stored on CRAN. There is a workaround - download dependency source files and compile into binaries locally. It's doable, but probably not a good idea. I am leaning towards dropping support for package versions.

**Comment by trybik** [comment]
After playing with the electron variant, doing some testing and fixes (pull request will come soon), I think now that including R installation local to the app with all packages pre-installed would be best. I've asked around and apparently Win R binary is portable (at least to some wide extent, like 64-bit systems).

**Comment by chasemc** [comment]
RE "apparently Win R binary is portable"...  definitely haven't  tested thoroughly, but I've transported between Win10 and Win XP before with success
